### PR TITLE
Add backup-extenstion for azure provider

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:
This PR adds the backup-extension support for provider-azure
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Please refer to #123.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: controllers/provider-aws/pkg/controller/backupentry/actuator.go [hunk 1]
```diff
@@ -53,5 +53,5 @@ func (a *actuator) Delete(ctx context.Context, be *extensionsv1alpha1.BackupEntr
 		return err
 	}
 
-	return awsClient.DeleteObjectsWithPrefix(ctx, be.Spec.BucketName, fmt.Sprintf("%s/", be.Name))
+	return awsClient.DeleteObjectsWithPrefix(ctx, be.Spec.BucketName, path.Join(be.Name, "/"))
```
Comment (rfranzke) [inline controllers/provider-aws/pkg/controller/backupentry/actuator.go]: Is this change intended? It leads to different results:

* `fmt.Sprintf("%s/", be.Name)` results in `<name>/`
* `path.Join(be.Name, "/")` results in `<name>`

See https://play.golang.com/p/1thfsZeX5J0

File: controllers/provider-azure/pkg/azure/types.go [hunk 2]
```diff
@@ -46,6 +47,16 @@ const (
...
+	//TOREMOVE: To keep up with naming convention we will change these constant related to storage. But will removed at the time we completely move to
+	// shared bucket approach.
+
+	// NewStorageAccount is a constant for the key in a cloud provider secret and backup secret that holds the Azure account name.
+	NewStorageAccount = "storageAccount"
+	// NewStorageKey is a constant for the key in a cloud provider secret and backup secret that holds the Azure secret storage access key.
+	NewStorageKey = "storageKey"
+
+	// AzureBlobStorageHostName is the host name for azure blob storage service.
+	AzureBlobStorageHostName = "blob.core.windows.net"
```
Comment (rfranzke) [inline controllers/provider-azure/pkg/azure/types.go]: Can't this be imported from the Azure library somewhere?

File: controllers/provider-aws/pkg/controller/backupentry/actuator.go (hunk 1 above)
Comment (swapnilgm) [reply controllers/provider-aws/pkg/controller/backupentry/actuator.go]: Ahh will revert that. Thanks for pointing that out.

File: controllers/provider-azure/pkg/azure/types.go (hunk 2 above)
Comment (swapnilgm) [reply controllers/provider-azure/pkg/azure/types.go]: Unfortunately not. I looked for it. I couldn't find it, even i was surprised to not see the constant defined for this. If you find it, i'll be happy to use it.

**Comment by swapnilgm** [comment]
Oops. I remember revendoring it. Will upload vendor part with corrections.

File: controllers/provider-azure/charts/provider-azure/templates/rbac.yaml [hunk 3]
```diff
@@ -9,11 +9,15 @@ rules:
 - apiGroups:
   - extensions.gardener.cloud
   resources:
+  - backupbucket
+  - backupbucket/status
+  - backupentry
+  - backupentry/status
```
Comment (rfranzke) [inline controllers/provider-azure/charts/provider-azure/templates/rbac.yaml]: Use

```yaml
  - backupbuckets
  - backupbuckets/status
  - backupentries
  - backupentries/status
```

like in https://github.com/gardener/gardener-extensions/commit/1cbdc1d07e75fb6f5a978b33c27c2127016e6ce6
